### DISCUSSION TITLE
Add a setting for switching day/night temperatures on widget

### DISCUSSION THREAD

**Discussion by A-Nicoladie** [breezy-weather/breezy-weather#1070] [Ideas]
### Describe your idea

Hi,
would it be possible to add a setting that allows to reverse the temperature display on the widget?

Switch from displaying [Day/Night] to [Night/Day]

Thanks !

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.3 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
The night temperature is the one from the overnight (from day N to day N+1), not from the previous night.

So currently it displays:
Daytime temperature (day N)/overnight (from day N to day N+1) temperature

I have no plans to replace it with:
Previous overnight (day N-1 to day N) temperature/daytime temperature (day N)

- the data from the previous day may be lacking
- it's also complex to handle because each day and overnight are stored on the day that is currently displayed and I see no great added value VS complexity
- we would also need to swap all other night values (such as weather icon)

**Comment by A-Nicoladie** [comment]
Sorry, maybe I misspoke: I was just talking about reversing the temperature display, not changing the values. 

Being able to switch between :
Daytime temperature (day N)/overnight (from day N to day N+1) temperature
And
Overnight (from day N to day N+1) temperature / Daytime temperature (day N)

👋

**Reply by papjul** [reply]
Why do you want to show a temperature that will come after *before* the one that will come first? :thinking:

**Comment by A-Nicoladie** [comment]
Because Habits die hard :grin: 
I used [GeometricWeather](https://github.com/WangDaYeeeeee/GeometricWeather) for a very long time and the temperature on my phone was displayed like that.
What's more, at least on most French weather sites, the display is often like this.
The display is read as the day's min/max and not as a timeline.


Example:

https://meteofrance.com/previsions-meteo-france/paris-1er-arrondissement/75001
![image](https://github.com/breezy-weather/breezy-weather/assets/90625228/48c17ce7-ec43-4f66-bf71-0ee6e5abb8df)

https://meteo.tf1.fr/meteo-france/ile-de-france/paris/paris-1er-arrondissement
![image](https://github.com/breezy-weather/breezy-weather/assets/90625228/ad05650e-4a45-4dd5-9e1f-10e9bd4bd6c9)

https://www.meteoconsult.fr/previsions-meteo/bulletin-detaille/ville-33/previsions-meteo-paris-aujourdhui
![image](https://github.com/breezy-weather/breezy-weather/assets/90625228/6df49ef2-bea4-4204-893f-f8b764d27ab0)

**Reply by papjul** [reply]
> The display is read as the day's min/max and not as a timeline.

Then, it makes totally sense these websites display minimum temperature of the day (from 00:00 to 23:59) before, because the minimum temperature is usually at 05:00~06:00 the night before in France. (But sometimes it can on rare occasions be at 23:00 instead, which makes the reading even more confusing, as you don't know on which night you will get cold, that's why daytime (from 06:00 to 17:59)/nighttime (from 18:00 to 29:59) temperature is much better IMO, especially as when looking at today weather, I don't care how cold it was last night, I want to know tonight).

Other websites with the same behavior as Breezy Weather also display in the same order as we do: https://www.accuweather.com/fr/fr/amiens/136096/daily-weather-forecast/136096

Swapping the order would make no sense and would give you the wrong impression that you're getting the same displayed info as your websites (even with swapped order, we still get reports from time to time "the min temperature is incorrect compared to website X"! Yes, that's because that's not the same data!). That's why this option was removed from Geometric Weather.

The only way it would make sense would be to display the night before first, as suggested earlier, but too much work and complexity.

Tl;dr: other websites use min/max, we use daytime/nighttime, not the same data ;)

**Comment by A-Nicoladie** [comment]
I understand your point of view (especially if I'm the only one who wants such an option 😁). 
I'll get used to it  👍

Thanks for your work.
